pandas/core/arrays/datetimes.py

```diff
@@ -291,7 +294,7 @@ def _from_sequence_not_strict(
         dtype=None,
         copy: bool = False,
         tz=None,
-        freq=lib.no_default,
+        freq: str | BaseOffset | lib.NoDefault | None = lib.no_default,
```

Comment:
Looks like `str` reappeared here

Reply:
Thanks, I will look into that! I had previously changed only `_simple_new` but not `_from_sequence_not_strict`.

Reply:
Oh sorry! From the diff I thought this was still under `_simple_new`, I think it's okay if freq is a `str` here